.github/workflows/pr.yaml

```diff
@@ -26,9 +26,47 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check-image:
+    name: Check Image & Changes
+    runs-on: ubuntu-latest
+    outputs:
+      relevant_changes: ${{ steps.check-image.outputs.relevant_changes }}
+      image_exists: ${{ steps.check-image.outputs.image_exists }}
+    steps:
+      - name: Run image & changes check
+        uses: ./.github/actions/check-image-and-changes
+
+  linting:
+    name: Run Linting & Prettier
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        uses: backstage/actions/yarn-install@v0.6.17
+        with:
+          cache-prefix: ${{ runner.os }}-v22
+
+      - name: Run Prettier
+        run: yarn prettier:check --continue --affected
+
+      - name: Run Lint
+        run: yarn run lint:check --continue --affected
+
   build:
     name: Build with Node.js ${{ matrix.node-version }}
```

Comment:
this is a required test. Can we run the step but have it immediately exit 0 if there's nothing to do?

![image](https://github.com/user-attachments/assets/7e0f733f-d703-4a54-9bf7-00a550565f97)